overlord/devicestate/handlers_install.go

```diff
@@ -314,11 +316,20 @@ func checkEncryption(model *asserts.Model) (res bool, err error) {
 
 	// encryption is required in secured devices and optional in other grades
 	if err := secbootCheckKeySealingSupported(); err != nil {
-		if secured {
+		if encrypted {
 			return false, fmt.Errorf("cannot encrypt secured device: %v", err)
```

Comment:
I suppose we need to tweak this message a bit:  maybe two variants:

grade == secured

cannot encrypt device storage as mandated by model grade secured

else

cannot encrypt device storage as mandated by encrypted storage-safety model option